qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,591 @@
+    """
+    b_p = _edge_operator_bi(edge_list, p)
+    b_q = _edge_operator_bi(edge_list, q)
+    id_op = _pauli_id(edge_list.shape[1])
+    qubit_op = (id_op - b_p) * (id_op - b_q)
+    if p == s:  # two commutations to order as two number operators.
+        final_coeff = 0.25
+    else:  # one commutation
+        final_coeff = -0.25
+    return (final_coeff * h2_pqrs) * qubit_op
+
+
+## SW2018 eq 35
+## Includes contributions from a h.c. pair
+def _excitation_operator(  # pylint: disable=invalid-name
+    edge_list: np.ndarray, p: int, q: int, h1_pq: float
+) -> SparsePauliOp:
+    """Map an excitation operator to a Pauli operator.
+
+    Args:
+      edge_list: representation of graph specifying neighboring qubits.
+      p: First Fermionic-mode index.
+      q: Second Fermionic-mode index. You must ensure that p < q.
+      h1_pq: Numerical coefficient of the term.
+
+    Returns:
+      The result of the Fermionic to Pauli operator mapping.
+    """  # pylint: disable=missing-raises-doc
+    if p >= q:
+        raise ValueError("Expected p < q, got p = ", p, ", q = ", q)
+    b_a = _edge_operator_bi(edge_list, p)
+    b_b = _edge_operator_bi(edge_list, q)
+    a_ab = _edge_operator_aij(edge_list, p, q)
+    return (-1j * 0.5 * h1_pq) * ((b_b & a_ab) + (a_ab & b_a))
+
+
+## SW2018 eq 37
+def _double_excitation(  # pylint: disable=invalid-name
+    edge_list: np.ndarray, p: int, q: int, r: int, s: int, h2_pqrs: float
+) -> SparsePauliOp:
+    """Map a double-excitation operator to a Pauli operator.
+
+    Args:
+      edge_list: representation of graph specifying neighboring qubits.
+      p: First Fermionic-mode index.
+      q: Second Fermionic-mode index. You must ensure that p < q.
+      r: Third Fermionic-mode index.
+      s: Fourth Fermionic-mode index.
+      h2_pqrs: Numerical coefficient of the term.
+
+    Returns:
+      The result of the Fermionic to Pauli operator mapping.
+    """
+    b_p = _edge_operator_bi(edge_list, p)
+    b_q = _edge_operator_bi(edge_list, q)
+    b_r = _edge_operator_bi(edge_list, r)
+    b_s = _edge_operator_bi(edge_list, s)
+    a_pq = _edge_operator_aij(edge_list, p, q)
+    a_rs = _edge_operator_aij(edge_list, r, s)
+    a_pq = -a_pq if q < p else a_pq
+    a_rs = -a_rs if s < r else a_rs
+
+    id_op = _pauli_id(edge_list.shape[1])
+    qubit_op = (a_pq * a_rs) * (
+        -id_op
+        - b_p * b_q
+        + b_p * b_r
+        + b_p * b_s
+        + b_q * b_r
+        + b_q * b_s
+        - b_r * b_s
+        + b_p * b_q * b_r * b_s  ## Agrees with SW2018 eq 37 and OpenFermion. Aqua had `-`.
+    )
+    final_coeff = 0.125
+    return (final_coeff * h2_pqrs) * qubit_op
+
+
+def _number_excitation(  # pylint: disable=invalid-name
+    edge_list: np.ndarray, p: int, q: int, r: int, s: int, h2_pqrs: float
+) -> SparsePauliOp:
+    """Map a number-excitation operator to a Pauli operator.
+
+    Exactly two of the indices p, q, r, s must be equal. These represent
```

Comment:
In looking at `q` below I guess this statement here about two of them being equal eliminates the pair p+q; I guess its fine without having that constraint listed here too in this pairing - assuming this is correct.

Reply:
Yes, it's correct. I could be even more specific about the requirement.